include/espi.h

```diff
@@ -0,0 +1,829 @@
+ *     |              |----------->+    response       |             |
+ *     |              |            |------------------>+             |
+ *     |              |            |                   |             |
+ *     |              |            | SET_CONFIGURATION |             |
+ *     |              |            +<------------------+             |
+ *     |              |            |  accept           |             |
+ *     |              |            +------------------>+             |
+ *     +              +            +                   +             +
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param cfg the device runtime configuration for the eSPI controller.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input / output error, failed to configure device.
+ * @retval -EINVAL invalid capabilities, failed to configure device.
+ * @retval -ENOTSUP capability not supported by eSPI slave
+ */
+__syscall int espi_config(struct device *dev, struct espi_cfg *cfg);
+
+static inline int z_impl_espi_config(struct device *dev,
+					struct espi_cfg *cfg)
+{
+	const struct espi_driver_api *api =
+		(const struct espi_driver_api *)dev->driver_api;
+
+	return api->config(dev, cfg);
+}
+
+/**
+ * @brief Query to see if it a channel is ready.
+ *
+ * This routine allows to check if logical channel is ready before use.
+ * Note that queries for channels not supported will always return false.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ch the eSPI channel for which status is to be retrieved.
+ *
+ * @retval true If eSPI channel is ready.
+ * @retval false otherwise.
+ */
+__syscall bool espi_get_channel_status(struct device *dev,
+	enum espi_channel ch);
+
+static inline bool z_impl_espi_get_channel_status(struct device *dev,
+							enum espi_channel ch)
+{
+	const struct espi_driver_api *api =
+		(const struct espi_driver_api *)dev->driver_api;
+
+	return api->get_channel_status(dev, ch);
+}
+
+/**
+ * @brief Sends memory, I/O or message read request over eSPI.
+ *
+ * This routines provides a generic interface to send a read request packet.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param req Structure representing a memory, I/O or message read request.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP if eSPI controller doesn't support raw packets and instead
+ *         low memory transactions are handled by controller hardware directly.
+ * @retval -EIO General input / output error, failed to send over the bus
+ */
+__syscall int espi_read_request(struct device *dev,
+					struct espi_request_packet req);
+
+static inline int z_impl_espi_read_request(struct device *dev,
+						struct espi_request_packet req)
+{
+	const struct espi_driver_api *api =
+		(const struct espi_driver_api *)dev->driver_api;
+
+	if (!api->read_request) {
+		return -ENOTSUP;
+	}
+	return api->read_request(dev, req);
```

Comment:
have an empty line after a '}' unless it's another '}' etc... Apply that everywhere.

Reply:
@tbursztyka  I re-checked and space guideline after if/else blocks is followed everywhere

Reply:
@tbursztyka can you please limit awarding patches -1s to *egregious* style issues, and incorrect code?
We're not going to make progress if we're going to sit here arguing about what blank line goes where

Reply:
I thought these comments were already applied long ago. The ones that were not applied were enum names and DTS stuff (which is a valid issue to -1)